### PR TITLE
style: require explicit device and dtype

### PR DESCRIPTION
**Pull request by njzjz**
Add a linter to require explicit `device` and `dtype` for `{np,tf,pt}.{tensor,zeros,ones,arange,range,empty,full,rand,eye,linspace}`.

- EDP01: Require explicit device when initializing a PyTorch tensor.
- EDP02: Require explicit dtype when initializing a NumPy array, a TensorFlow tensor, or a PyTorch tensor.

Missing `device` or `dtype` often causes a problem.

Note: I am not going to fix the existing codes in this PR, as too many (200+) lines violate the rules. Fixing them takes a long time which I don't have. I disable rules on those lines.

## Summary by CodeRabbit

- **New Features**
	- Added enhanced linting capabilities through a new `deepmd_checker` plugin in the `pyproject.toml` configuration.
  
- **Bug Fixes**
	- Improved code clarity and compliance by suppressing pylint warnings related to explicit data types across multiple files.

- **Documentation**
	- Numerous comments added to clarify the suppression of linting messages, enhancing code maintainability and readability.

### CODE REVIEW

File: deepmd/tf/nvnmd/entrypoints/mapt.py [hunk 1]
```diff
@@ -537,11 +537,11 @@
         keys = list(dic_ph.keys())
         vals = list(dic_ph.values())
 
-        s = N2 * np.reshape(np.arange(0, N + 1) / N, [-1, 1]) + smin_
+        s = N2 * np.reshape(np.arange(0, N + 1) / N, [-1, 1]) + smin_  # pylint: disable=no-explicit-dtype
```
[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/tf/nvnmd/entrypoints/mapt.py:540]: It seems right but not related to this PR.

File: deepmd/tf/fit/polar.py [hunk 2]
```diff
@@ -190,7 +190,7 @@
 
         # YWolfeee: support polar normalization, initialize to a more appropriate point
         if self.shift_diag:
-            mean_polar = np.zeros([len(self.sel_type), 9])
+            mean_polar = np.zeros([len(self.sel_type), 9])  # pylint: disable=no-explicit-dtype
             sys_matrix, polar_bias = [], []
```
[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/tf/fit/polar.py:194]: It seems right but not related to this PR.

File: deepmd/tf/descriptor/se_a_mask.py [hunk 3]
```diff
@@ -319,9 +322,9 @@
 
         with tf.variable_scope("descrpt_attr" + suffix, reuse=reuse):
             if davg is None:
-                davg = np.zeros([self.ntypes, self.ndescrpt])
+                davg = np.zeros([self.ntypes, self.ndescrpt])  # pylint: disable=no-explicit-dtype
```
[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/tf/descriptor/se_a_mask.py:325]: It seems right but not related to this PR.

File: deepmd/tf/descriptor/se_a_mask.py [hunk 4]
```diff
@@ -179,13 +179,16 @@
...
+                ii * np.ones(self.sel_a[ii]),  # pylint: disable=no-explicit-dtype
+            )  # like a mask
         self.nei_type = tf.constant(nei_type, dtype=tf.int32)
 
-        avg_zero = np.zeros([self.ntypes, self.ndescrpt]).astype(
+        avg_zero = np.zeros([self.ntypes, self.ndescrpt]).astype(  # pylint: disable=no-explicit-dtype
             GLOBAL_NP_FLOAT_PRECISION
         )
-        std_ones = np.ones([self.ntypes, self.ndescrpt]).astype(
+        std_ones = np.ones([self.ntypes, self.ndescrpt]).astype(  # pylint: disable=no-explicit-dtype
```
[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/tf/descriptor/se_a_mask.py:191]: It seems right but not related to this PR.

File: deepmd/tf/descriptor/se_a_mask.py [hunk 5]
```diff
@@ -319,9 +322,9 @@
 
         with tf.variable_scope("descrpt_attr" + suffix, reuse=reuse):
             if davg is None:
-                davg = np.zeros([self.ntypes, self.ndescrpt])
+                davg = np.zeros([self.ntypes, self.ndescrpt])  # pylint: disable=no-explicit-dtype
             if dstd is None:
-                dstd = np.ones([self.ntypes, self.ndescrpt])
+                dstd = np.ones([self.ntypes, self.ndescrpt])  # pylint: disable=no-explicit-dtype
```
[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/tf/descriptor/se_a_mask.py:327]: It seems right but not related to this PR.

File: deepmd/tf/descriptor/se_a_mask.py [hunk 6]
```diff
@@ -179,13 +179,16 @@
...
         for ii in range(self.ntypes):
-            nei_type = np.append(nei_type, ii * np.ones(self.sel_a[ii]))  # like a mask
+            nei_type = np.append(
+                nei_type,
+                ii * np.ones(self.sel_a[ii]),  # pylint: disable=no-explicit-dtype
+            )  # like a mask
         self.nei_type = tf.constant(nei_type, dtype=tf.int32)
 
-        avg_zero = np.zeros([self.ntypes, self.ndescrpt]).astype(
+        avg_zero = np.zeros([self.ntypes, self.ndescrpt]).astype(  # pylint: disable=no-explicit-dtype
```
[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/tf/descriptor/se_a_mask.py:188]: It seems right but not related to this PR.

File: deepmd/tf/nvnmd/entrypoints/mapt.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/tf/nvnmd/entrypoints/mapt.py:540]: It seems right but not related to this PR.

File: deepmd/pt/model/descriptor/se_a.py [hunk 7]
```diff
@@ -635,7 +635,7 @@
             dmatrix = dmatrix.view(
                 -1, self.ndescrpt
             )  # shape is [nframes*nall, self.ndescrpt]
-            xyz_scatter = torch.empty(
+            xyz_scatter = torch.empty(  # pylint: disable=no-explicit-dtype
```
[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/pt/model/descriptor/se_a.py:638]: It seems right but not related to this PR.

**Comment by njzjz** [comment]
CodeQL alerts are not related to this PR (but it's expected to fix them in a separated PR)

**Review by wanghan-iapcm — APPROVED** [review]

**Review by iProzd — APPROVED** [review]

**Pull request merged**